sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCRelation.scala

```diff
@@ -174,10 +174,21 @@ private[sql] object JDBCRelation extends Logging {
     (dialect.quoteIdentifier(column.name), column.dataType)
   }
 
-  private def toInternalBoundValue(value: String, columnType: DataType): Long = columnType match {
-    case _: NumericType => value.toLong
-    case DateType => DateTimeUtils.fromJavaDate(Date.valueOf(value)).toLong
-    case TimestampType => DateTimeUtils.fromJavaTimestamp(Timestamp.valueOf(value))
+  private def toInternalBoundValue(
+      value: String,
+      columnType: DataType,
+      timeZoneId: String): Long = {
+    def parse[T](f: UTF8String => Option[T]): T = {
+      f(UTF8String.fromString(value)).getOrElse {
```

Comment:
Is this pattern being repeated? We could add a util somewhere and reuse it. We can do it in another place assuming you're still working about this here and there.

Reply:
There is a function there https://github.com/apache/spark/pull/23541/files#diff-9847f5cef7cf7fbc5830fbc6b779ee10R1555 which applies `UTF8String.fromString` and `f: F8String => Option[T]` but this is the only common code. Do you think is it worth to extract the common code?

Reply:
I see that's only repeated twice. It's okie. Let's extract it later if it starts to be repeated :).